api/opds_for_distributors.py

```diff
@@ -119,12 +119,14 @@ def refresh(credential):
                 raise LibraryAuthorizationFailedException()
             credential.credential = access_token
             expires_in = expires_in
-            credential.expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=expires_in)
+            # We'll avoid edge cases by assuming the token expires 75%
+            # into its useful lifetime.
+            credential.expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=expires_in*0.75)
         return Credential.lookup(_db, self.data_source_name,
                                  "OPDS For Distributors Bearer Token",
                                  patron=None,
                                  refresher_method=refresh,
-                                 ).credential
+                                 )
```

Comment:
We return the Credential object instead of its string because we need all the information in the Credential object to generate the token document.